doc/devdocs/common/FilePreviewCommon.md

```diff
@@ -0,0 +1,70 @@
+
+```javascript
+export function idDefinition() {
+    return {
+        ...
+    }
+}
+```
+
+2. Add the following line to the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/monacoSpecialLanguages.js) file:
+
+```javascript
+import { idDefinition } from './customLanguages/file.js';
+```
+
+3. In the [`monacoSpecialLanguages.js`](/src/modules/previewpane/MonacoPreviewHandler/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function:
+
+```javascript
+registerAdditionalNewLanguage("id", [".fileExtension"], idDefinition(), monaco)
+```
+
+  * The id can be anything. Recommended is one of the file extensions. For example "php" or "reg".
+4. Copy the existing language definition into the `languageDefinitions` function in the same file. You can find the existing definitions in the following folder: [`/src/common/FilePreviewCommon/monacoSRC/min/vs/basic-languages`](/src/common/FilePreviewCommon/monacoSRC/min/vs/basic-languages).
+
+5. Execute the steps described in the [Generate monaco_languages.json file](#monaco_languagesjson) section.
+
+### Add a new file extension to an existing language
+
+1. In the [`monacoSpecialLanguages.js`](/src/common/FilePreviewCommon/monacoSpecialLanguages.js) file add the following line to the `registerAdditionalLanguages` function (`existingId` is the id of the language you want to add the extension to. You can find these id's in the `monaco_languages.json` file)):
+
+```javascript
+registerAdditionalLanguage("id", [".fileExtension"], "existingId", monaco)
+```
+
+  * If for instance you want to add more extensions to the php language set the id to `phpExt` and the existingId to `php`.
+
+2. Execute the steps described in the [Generate monaco_languages.json file](#monaco_languagesjson) section.
+
+### monaco_languages.json
+
+[`monaco_languages.json`](/src/common/FilePreviewCommon/monaco_languages.json) contains all extensions and Id's for the languages supported by Monaco. The [`MonacoHelper`](/src/common/FilePreviewCommon/MonacoHelper.cs) class and the installer are using this file to register preview handlers for defined extensions.
+
+After updating Monaco Editor and/or adding a new language you should update the [`monaco_languages.json`](/src/common/FilePreviewCommon/monaco_languages.json) file.
+
+1. Run the file on a local webserver (as webbrowsers will block certain needed features when running the file locally.)
+  *  This can for example be achieved by using the [Preview Server](https://marketplace.visualstudio.com/items?itemName=yuichinukiyama.vscode-preview-server) extension for Visual Studio Code: Open the file in Visual Studio Code right click in the code editor and select `vscode-preview-server: Launch on browser`. The file will then be opened in a browser.
```

Comment:
```suggestion
  *  This can for example be achieved by using the [Preview Server](https://marketplace.visualstudio.com/items?itemName=yuichinukiyama.vscode-preview-server) extension for Visual Studio Code: Open the file in Visual Studio Code, right click in the code editor and select `vscode-preview-server: Launch on browser`. The file will be opened in a browser.
```